Presentation.Web/app/shared/main-contract/main-contract-section.component.ts

```diff
@@ -0,0 +1,101 @@
+﻿module Kitos.Shared.Components {
+    "use strict";
+
+    function setupComponent(): ng.IComponentOptions {
+        return {
+            bindings: {
+                mainText: "@",
+                hasWriteAccess: "=",
+                isVisible: "=",
+                viewModel: "<",
+            },
+            controller: MainContractSectionController,
+            controllerAs: "ctrl",
+            templateUrl: `app/shared/main-contract/main-contract-section.view.html`
+        };
+    }
+
+    export interface IMainContractSectionViewModel {
+        isActive: boolean;
+        selectedContractId: number;
+        options: Models.ViewModel.Generic.Select2OptionViewModel<null>[];
+        postMethod: (contractId: number) => ng.IPromise<void>;
+        deleteMethod: () => ng.IPromise<void>;
+        stateReloadMethod: () => ng.IPromise<void>;
+    }
+    
+    interface IMainContractSectionController extends ng.IComponentController {
+        mainText: string;
+        hasWriteAccess: boolean;
+        isVisible: boolean;
```

Comment:
hm... since it's an element in the html, the parent component could just add an ng-if in stead of sending it to the sub component thereby complicating the config..